applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationAsyncClient.java

```diff
@@ -94,9 +96,19 @@ public static ConfigurationAsyncClientBuilder builder() {
 
         // This service method call is similar to setSetting except we're passing If-Not-Match = "*". If the service
         // finds any existing configuration settings, then its e-tag will match and the service will return an error.
+        // Note: addSetting is a special case where its precondition failed exception is the inverse of all other
+        // requests that route through the client.setKey method, remap it to a more appropriate error.
         return service.setKey(serviceEndpoint, setting.key(), setting.label(), setting, null, getETagValue(ETAG_ANY))
             .doOnRequest(ignoredValue -> logger.asInformational().log("Adding ConfigurationSetting - {}", setting))
             .doOnSuccess(response -> logger.asInformational().log("Added ConfigurationSetting - {}", response.value()))
+            .onErrorMap(error -> {
+                if (!(error instanceof ResourceNotFoundException)) {
+                    return error;
+                }
+
+                ResourceNotFoundException notFoundException = (ResourceNotFoundException) error;
+                return new ResourceModifiedException(notFoundException.getMessage(), notFoundException.response());
+            })
```

Comment:
It feels like this should be a utility method?

Reply:
Done